lib/link.js

```diff
@@ -118,7 +118,11 @@ Link.prototype.detach = function(options) {
     self.once(Link.Detached, function(info) {
       self.removeListener(Link.ErrorReceived, onError);
       if (!!info.error) return reject(info.error);
-      if (!info.closed) return reject('link not closed');
+      if (!info.closed) {
+        if (!options || options.closed) {
+          return reject('link not closed');
```

Comment:
don't we actually want to `resolve` in this case, as the detach has successfully occurred even though the link was not closed?